### PR TITLE
feature/199

### LINKED ISSUE #199: Airport Revisit: Frankfurt Airport
- Add remaining transitions  
- Add more entry points per transition

All work done by @jakobeng1303!

Seen in zlsa/atc#733.

### PR DESCRIPTION
**Pull request by jakobeng1303**
Resolves #199.

Sadly I don't have a lot of time right now due to my upcoming finals. I more or less copied all the changes from the old file (which I proposed a while ago) and made compatible. Expect to have some typos or mistakes included :( To allow the player to use different transitions, the airspace has to be changed slightly (I think!)

### CODE REVIEW

File: assets/airports/eddf.json [hunk 1]
```diff
@@ -379,111 +379,239 @@
...
-      "rwy": {
+      "ROLIS7N": {
+        "icao": "ROLIS7N",
+        "name": "Rolis Seven North",
+        "entryPoints": {
+          "ROLIS": [["ROLIS", "A150-"], ["ETARU", "A100|S250"]],
+          "ETARU": [["ETARU", "A100|S250"]],
+          "LIPMI": ["LIPMI", ["COL", "A230-"], ["ROLIS", "A150-"], ["ETARU", "A100|S250"]]
+        },
+        "body": [["DF432", ["DF439", "A80"], ["DF441", "A50"], "DF442", "DF443", "DF444", "DF454", "DF453", "DF452"],
```
Comment (Fechulo) [inline assets/airports/eddf.json]: There's an extra square bracket on line 465

Comment (jakobeng1303) [inline assets/airports/eddf.json]: thanks!

**Comment by jakobeng1303** [comment]
@indianbhaji I din't change any of these values in the process. There are no speed restrictions though so why change it? Is that a problem of the game?

**Comment by indianbhaji** [comment]
@jakobeng1303 If you know that definitely there are no propeller aircraft, it's fine however it will be likely that there will be propeller aircraft. Propeller aircraft will not be able to catch up to speed and will cause conflicts with larger aircraft.

**Comment by jakobeng1303** [comment]
@indianbhaji except for the Dh8D there shouldn't be any turboprops! (the Dash can fly 320!) https://github.com/openscope/openscope/blob/develop/assets/aircraft/dh8d.json

File: assets/airports/eddf.json [hunk 2]
```diff
@@ -379,111 +379,239 @@
...
+        "body": ["CHA", ["DF606", "A80"], "DF610", ["DF611", "A40"], "DF612", "DF613", "DF614", "DF615", "DF616", "DF626", "DF625", "DF624", "DF623", "DF622", "DF621"],
+        "rwy": {
+        "25L": [["LEDKI", "A40"]],
+        "25C": [["REDGO", "A40"]],
+        "25R": []
+        }
       },
-      "body": [["PSA", "A110|S250"], "CHA", "DF606", "DF610", "DF611", "DF612", "DF613", "DF614", "DF615", "DF616", "DF626", "DF625", "DF624", "DF623", "DF622", "DF621", "LEDKI"],
-      "rwy": {
+      "PSA25N": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

File: assets/airports/eddf.json [hunk 3]
```diff
@@ -379,111 +379,239 @@
...
-    "KERAX25N": {
-      "icao": "KERAX25N",
-      "name": "Kerax Two Five North",
-      "entryPoints": {
-        "GAPLA": ["GAPLA", ["DEMAB", "A160-"], ["SOLVU", "A130-"]]
+        }
       },
-      "body": [["KERAX", "A110|S250"], "GED", "DF407", "DF408", "DF411", "DF412", "DF413", "DF414", "DF415", "DF416", "DF426", "DF425", "DF424", "DF423", "DF422", "DF421"],
-      "rwy": {
+      "PSA7N": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

File: assets/airports/eddf.json [hunk 4]
```diff
@@ -379,111 +379,239 @@
...
-      "name": "Kerax Seven South",
-      "entryPoints": {
-        "GAPLA": ["GAPLA", ["DEMAB", "A190-"], ["SOLVU", "A160-"]]
+        }
       },
-      "body": [["KERAX", "A140-"], ["EBIPA", "A120|S250"], "INBOS", "PSA", "CHA", "DF635", "DF636", "DF640", "DF641", "DF642", "DF643", "DF644", "DF654", "DF653", "DF652", "DF651", "ROBSA"],
-      "rwy": {
-        "07L": [],
-        "07C": [],
+      "PSA7S": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

File: assets/airports/eddf.json [hunk 5]
```diff
@@ -379,111 +379,239 @@
...
-        "07C": [],
-        "07R": []
-      }
-    },
-    "ROLIS25N": {
-      "icao": "ROLIS25N",
-      "name": "Rolis Two Five North",
-      "entryPoints": {
-        "LIPMI": ["LIPMI", ["COL", "A230-"]]
+      "KERAX25S": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

File: assets/airports/eddf.json [hunk 6]
```diff
@@ -379,111 +379,239 @@
...
+          "GAPLA": ["GAPLA", ["DEMAB", "A160-"], ["SOLVU", "A130-"], ["KERAX", "A110|S250"]]
+        },
+        "body": [["EBIPA", "A120|S250"], "INBOS", "PSA", "CHA", "DF635", ["DF636", "A80"], "DF640", ["DF641", "A50"], "DF642", "DF643", "DF644", "DF654", "DF653", "DF652", "DF651"],
+        "rwy": {
+        "07L": [["ROBSA", "A40"]],
+        "07C": [["LOMPO", "A40"]],
+        "07R": []
+        }
+      },
+      "KERAX7N": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

File: assets/airports/eddf.json [hunk 7]
```diff
@@ -379,111 +379,239 @@
...
+          "LIPMI": ["LIPMI", ["COL", "A230-"], ["ROLIS", "A150-"], ["ETARU", "A100|S250"]]
+        },
+        "body": ["DF403", "DF409", "DF410", ["DF411", "A50"], "DF412", "DF413", "DF414", "DF415", "DF416", "DF426", "DF425", "DF424", "DF423", "DF422", "DF421"],
+        "rwy": {
+          "25L": [],
+          "25C": [],
+          "25R": []
+        }
+      },
+      "KERAX7S": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

File: assets/airports/eddf.json [hunk 8]
```diff
@@ -379,111 +379,239 @@
...
+          "LIPMI": ["LIPMI", ["COL", "A230-"], ["ROLIS", "A150-"], ["ETARU", "A100|S250"]]
+        },
+        "body": ["DF432", ["DF439", "A80"], ["DF441", "A50"], "DF442", "DF443", "DF444", "DF454", "DF453", "DF452"],
+        "rwy": {
+          "07L": [],
+          "07C": [],
+          "07R": []
+        }
+      },
+      "ROLIS25S": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

File: assets/airports/eddf.json [hunk 9]
```diff
@@ -379,111 +379,239 @@
...
-    "UNOKO25S": {
-      "icao": "UNOKO25S",
-      "name": "Unoko Two Five South",
-      "entryPoints": {
-        "NIVNU": ["NIVNU", ["RASVO", "A280-"], ["PIPEP", "A250-"]]
+        }
       },
-      "body": [["UNOKO", "A230-"], ["IBVIL", "A210-"], ["MANUV", "A180-"], ["RAMOB", "A130|S250"], "DF601", "DF609", "DF610", "DF611", "DF612", "DF613", "DF614", "DF615", "DF616", "DF626", "DF625", "DF624", "DF623", "DF622", "DF621", "LEDKI"],
-      "rwy": {
+      "ROLIS7N": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

File: assets/airports/eddf.json [hunk 10]
```diff
@@ -379,111 +379,239 @@
...
+          "NIVNU": ["NIVNU", ["RASVO", "A260-"], ["PIPEP", "A230-"], ["UNOKO", "A230-"], ["IBVIL", "A210-"], ["MANUV", "A180-"], ["RAMOB", "A130|S250"]]
+        },
+        "body": ["DF401", "DF403", ["DF409", "A80"], "DF410", ["DF411", "A50"], "DF412", "DF413", "DF414", "DF415", "DF416", "DF426", "DF425", "DF424", "DF423", "DF422", "DF421"],
+        "rwy": {
+          "25L": [],
+          "25C": [],
+          "25R": []
+        }
+      },
+      "ROLIS7S": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

File: assets/airports/eddf.json [hunk 11]
```diff
@@ -379,111 +379,239 @@
...
+          "NIVNU": ["NIVNU", ["RASVO", "A260-"], ["PIPEP", "A230-"], ["UNOKO", "A230-"], ["IBVIL", "A210-"], ["MANUV", "A180-"], ["RAMOB", "A130|S250"]]
+        },
+        "body": ["DF601", ["DF609", "A80"], "DF610", ["DF611", "A40"], "DF612", "DF613", "DF614", "DF615", "DF616", "DF626", "DF625", "DF624", "DF623", "DF622", "DF621"],
+        "rwy": {
+        "25L": [["LEDKI", "A40"]],
+        "25C": [["REDGO", "A40"]],
+        "25R": []
+        }
+      },
+      "UNOKO25N": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

File: assets/airports/eddf.json [hunk 12]
```diff
@@ -379,111 +379,239 @@
...
+          "07L": [["ROBSA", "A40"]],
+          "07C": [["LOMPO", "A40"]],
+          "07R": []
+        }
       },
-      "body": [["UNOKO", "A210-"], ["IBVIL", "A190-"], ["MANUV", "A160-"], ["RAMOB", "A110|S250"], "DF431", "DF432", "DF439", "DF441", "DF442", "DF443", "DF444", "DF454", "DF453", "DF452"],
-      "rwy": {
-        "07L": [],
-        "07C": [],
+      "UNOKO7N": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

File: assets/airports/eddf.json [hunk 13]
```diff
@@ -379,111 +379,239 @@
     }
   },
   "stars": {
-    "UNOKO7N": {
-      "icao": "UNOKO7N",
-      "name": "Unoko Seven North",
-      "entryPoints": {
-        "NIVNU": ["NIVNU", ["RASVO", "A260-"], ["PIPEP", "A230-"]]
+      "UNOKO7S": {
```
Comment (indianbhaji) [inline assets/airports/eddf.json]: You need to add this arrival to the spawnPatterns (it's no use otherwise)

**Comment by n8rzz** [comment]
@jakobeng1303 see the [spawnPatternReadme](https://github.com/openscope/openscope/blob/develop/src/assets/scripts/client/trafficGenerator/spawnPatternReadme.md) for more information on what, exactly, a `spawnPattern` is and how it should be structured.

**Comment by jakobeng1303** [comment]
@n8rzz thank you for the link! 
@indianbhaji I've added ALL the missing transition of which some are not be used with the current wind configuration. Maybe later the player will have an ATIS like field where you choose your desired conditions (just a random idea that came to my mind right now :D). 
All the other entry points are to be used by the player if he/she wants to re-clear a plane. 
As far as I think, the current star information that was used the old EDDF file is still active and _should_ work!

**Comment by n8rzz** [comment]
@jakobeng1303 have you had a chance to test it locally (see: https://github.com/openscope/openscope/tree/develop/tools)?  If everything is ok, you shouldn't see any console errors.  

If you are unable to test locally, I have spun up a review app where you will be able to run the code from your branch in a development environment.  Just click the `view deployment` buttom just a little further up the page.

**Comment by erikquinn** [comment]
Closing in favor of a local openscope branch, where I've reopened this PR.

See the branch `feature/199` and PR #255 for the replacement. I figure if he's busy, we can all kinda tagteam this thing and get EDDF sorted out. :smile: 

@jakobeng1303 Check your email for an invitation to the openscope team and you're welcome on the _slack_ team as well, where we all communicate, chat, and coordinate our work, etc. For the latter, I'd need an email address to send you the invite; you can shoot that to me at erikquinn2@gmail.com if you like. Whenever you get more time, feel free to come by and help out!

As always, if anybody wants invites, just advise!

**Pull request closed without merging**